### PR TITLE
QBD landing page

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

**Comment by DhaaraniCIT** [comment]
Qbd integration
component: onboarding-landing-page
https://user-images.githubusercontent.com/55541808/214527836-d2796442-adc2-4f4f-95d6-d2faab99296b.mov

File: src/app/core/services/qbd/qbd-core/qbd-workspace.service.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+export class QbdWorkspaceService {
+
+  constructor(
+    private apiService: QbdApiService
+  ) { }
+
+  qbdGetWorkspace(orgId: string): Observable<Workspace> {
```
Comment (ashwin1111) [inline src/app/core/services/qbd/qbd-core/qbd-workspace.service.ts:15]: `getQBDWorkspace`, same for post

File: src/app/integrations/qbd/onboarding/onboarding-landing/landing.component.html [hunk 2]
```diff
@@ -1 +1,21 @@
-<p>landing works!</p>
+<div class="qbd-landing">
```
Comment (ashwin1111) [inline src/app/integrations/qbd/onboarding/onboarding-landing/landing.component.html:1]: remove class

File: src/app/integrations/qbd/onboarding/onboarding.component.html [hunk 3]
```diff
@@ -1,2 +1,2 @@
-<app-qbd-header></app-qbd-header>
+<!-- <app-qbd-header></app-qbd-header> -->
```
Comment (ashwin1111) [inline src/app/integrations/qbd/onboarding/onboarding.component.html:1]: remove component as well

File: src/app/integrations/qbd/qbd.component.ts [hunk 4]
```diff
@@ -1,15 +1,78 @@
...
+    });
+  }
+
+  setupWorkspace() {
+    this.user = this.userService.qbdGetUserProfile();
+    this.getOrCreateWorkspace().then((workspace: Workspace) => {
+      this.workspace = workspace;
+      this.storageService.set('workspaceId', this.workspace.id);
+      this.storageService.set('onboardingState', 'Landing');
+      this.storageService.set('workspaceCreatedAt', workspace.created_at);
```
Comment (ashwin1111) [inline src/app/integrations/qbd/qbd.component.ts:68]: remove this

File: src/app/shared/components/apps/app-header/app-header.component.html [hunk 5]
```diff
@@ -2,32 +2,55 @@
...
+                <div class="tw-rounded-4-px tw-border-1-px tw-border-box-color tw-flex tw-justify-center tw-bg-white tw-w-176-px">
+                    <img src="assets/logos/bamboo-hr-logo.svg" width="160px" height="44px" class="tw-py-10-px tw-px-20-px" />
                 </div>
-                <p *ngIf="!isBambooSetupInProgress" class="tw-text-sub-text-color tw-pt-8-px tw-text-14-px">Auto sync employee details between Bamboo HR and Fyle.</p>
-                <div *ngIf="isBambooSetupInProgress" class="tw-flex tw-text-slightly-normal-text-color tw-text-14-px tw-pt-10-px tw-pb-18-px">
-                    <app-loader [styleClass]="'spinner-16 tw-top-2-px'"></app-loader>
-                    <p class="tw-pl-10-px">We are syncing your data between your Fyle - Bamboo HR account and this may take a few seconds</p>
+                <div class="tw-pl-16-px tw-flex tw-flex-col">
+                    <div class="tw-flex tw-pt-6-px">
+                        <p class="tw-text-20-px tw-font-500">BambooHR</p>
```
Comment (ashwin1111) [inline src/app/shared/components/apps/app-header/app-header.component.html:12]: `{{ appName }}`

File: src/app/shared/components/apps/app-header/app-header.component.html [hunk 6]
```diff
@@ -2,32 +2,55 @@
 <div class="tw-absolute tw-bg-header-2 tw-h-110-px tw-w-[100%] tw-z-[-1]"></div>
 
 <div class="tw-p-24-px tw-text-slightly-normal-text-color">
-    <div class="tw-h-68-px tw-flex tw-justify-between">
```
Comment (ashwin1111) [inline src/app/shared/components/apps/app-header/app-header.component.html:5]: retain old code, send boolean flags to disable loaders

File: src/app/shared/components/qbd/core/qbd-header/qbd-header.component.scss [hunk 7]
```diff
@@ -0,0 +1,8 @@
+.qbd-header {
+    background-color: #ffffff;
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/core/qbd-header/qbd-header.component.scss:2]: replace with tailwind styles

File: src/app/shared/components/apps/app-header/app-header.component.html [hunk 8]
```diff
@@ -3,32 +3,36 @@
...
-                <div *ngIf="isBambooSetupInProgress" class="tw-flex tw-text-slightly-normal-text-color tw-text-14-px tw-pt-10-px tw-pb-18-px">
-                    <app-loader [styleClass]="'spinner-16 tw-top-2-px'"></app-loader>
-                    <p class="tw-pl-10-px">We are syncing your data between your Fyle - Bamboo HR account and this may take a few seconds</p>
+                <div class="tw-pl-16-px tw-flex tw-flex-col">
+                    <div class="tw-flex tw-pt-6-px">
+                        <p class="tw-text-20-px tw-font-500">{{ appName }}</p>
+                        <svg-icon-sprite *ngIf="isIntegrationConnected && !isLoading && bambooHrConfiguration?.recipe_status && !hideRefreshIcon" pTooltip="Sync Data" tooltipPosition="right" src="sync" width="18px" height="14px" class="tw-cursor-pointer tw-ml-10-px tw-pt-6-px tw-text-faded-text-color" (click)="syncData()"></svg-icon-sprite>
+                    </div>
+                    <p *ngIf="!isIntegrationSetupInProgress" class="tw-text-sub-text-color tw-pt-8-px tw-text-14-px">{{ appDesc }}</p>
```
Comment (ashwin1111) [inline src/app/shared/components/apps/app-header/app-header.component.html:15]: `{{ appDescription }}`

File: src/app/integrations/qbd/qbd.component.ts [hunk 9]
```diff
@@ -1,15 +1,77 @@
...
+      });
+    });
+  }
+
+  setupWorkspace() {
+    this.user = this.userService.qbdGetUserProfile();
+    this.getOrCreateWorkspace().then((workspace: Workspace) => {
+      this.workspace = workspace;
+      this.storageService.set('workspaceId', this.workspace.id);
+      this.storageService.set('QBDOnboardingState', 'Landing');
```
Comment (ashwin1111) [inline src/app/integrations/qbd/qbd.component.ts:67]: add a // TODO comment to this to remove later, once API is ready

File: src/app/shared/components/apps/app-header/app-header.component.html [hunk 10]
```diff
@@ -3,32 +3,36 @@
 
 <div class="tw-p-24-px tw-text-slightly-normal-text-color">
     <div class="tw-h-68-px tw-flex tw-justify-between">
-        <div class="tw-flex">
```
Comment (ashwin1111) [inline src/app/shared/components/apps/app-header/app-header.component.html:6]: from the next PR, please don't format HTML code, it's very tough to review and also to identify which code has changed in the current PR

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**